image-layout.md

```diff
@@ -0,0 +1,55 @@
+## Open Container Initiative Image Layout Specification
+
+The OCI Image Layout is a slash separated layout of OCI content-addressable blobs and human significant [location-addressable](https://en.wikipedia.org/wiki/Content-addressable_storage#Content-addressed_vs._location-addressed) references (refs).
+This layout MAY be used in a variety of different transport mechanisms: archive formats (e.g. tar, zip), shared filesystem environments (e.g. nfs), or networked file fetching (e.g. http, ftp, rsync).
+Given an image layout a tool can convert a given ref into a runnable OCI Image Format by finding an appopriate manifest from the manifest list, unpacking the filesystem serializations in the correct order, and then converting the image configuration into an OCI Runtime config.json.
+
+The image layout has two top level directories:
+
+- "blobs" contains content-addressable blobs. A blob has no schema and should be considered opaque. It is addressed by its hashed contents from byte 0 to the end of the file.
+- "refs" contains descriptors pointing to an image manifest list
+
+It also contains a file that is used to identify the layout version:
+
+- "oci-layout" MUST contain a JSON object with a version field `{"version": "1.0.0"}` and MAY include additional fields.
+
+This is an example image layout:
+
+```
+$ cd example.com/app/
+$ find .
+.
+./blobs
+./blobs/sha256-afff3924849e458c5ef237db5f89539274d5e609db5db935ed3959c90f1f2d51
+./blobs/sha256-5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270
+./blobs/sha256-e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f
```

Comment:
On Tue, Jun 07, 2016 at 09:21:52PM -0700, Stephen Day wrote:
> ext4 is okay but many file systems do still struggle without the
> prefixed hash directories…

Under blob counts that you'd still consider stuffing into a single
tarball?  I'd still like to see someone post a link to a real-life
tarball that uses this layout to motivate any optimizations like this,
since the tar-layer approach makes me think it unlikely that you have
more than a few dozen blobs in a given layout.

> There are also cases where you want to partition based on algorithm…

Can you point to docs on those cases?